File: src/main/java/ssg/search/collection/es/ESCollection.java
package ssg.search.collection.es;

import QueryAPI510.Search;
import io.searchbox.core.SearchResult;
import ssg.search.base.Collection;
import ssg.search.parameter.Parameter;
import ssg.search.result.Result;

import java.util.List;

public abstract class ESCollection implements Collection {
	private String collectionName = "item";

	public Result getResult(SearchResult searchResult, String name, Parameter parameter, Result result){
		return null;
	}

	public List<String> getQuery(Parameter parameter) {
		return null;
	}

	public List<String> getOrQuery(Parameter parameter) {
		return null;
	}

	public void setCollectionName(String collectionName) {
		this.collectionName = collectionName;
	}

	@Override
	public String getCollectionName(Parameter parameter) {
		return collectionName;
	}

	@Override
	public String getCollectionAliasName(Parameter parameter) {
		return "ESCollection";
	}

	@Override
	public String[] getDocumentField(Parameter parameter) {
		return null;
	}

	@Override
	public String[] getSearchField(Parameter parameter) {
		return null;
	}

	@Override
	public Result getResult(Search search, String name, Parameter parameter, Result result) {
		return null;
	}
}

File: src/main/kotlin/ssg/front/search/api/collection/es/EsCollection.kt
package ssg.front.search.api.collection.es

import org.elasticsearch.action.search.SearchResponse
import ssg.front.search.api.dto.Parameter
import ssg.front.search.api.dto.result.Result

abstract class EsCollection {
    open var indexName = "item"

    abstract fun getQuery(parameter: Parameter): List<String>
    abstract fun getResult(searchResponse: SearchResponse, name: String, parameter: Parameter, result: Result)

    open fun getClassName(parameter: Parameter): String {
        return this::class.simpleName!!
    }
}